src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts

```diff
@@ -455,7 +471,19 @@ export class DatasetFormComponent implements Formconfiguration{
       entityForm.setDisabled('name',true);
       _.find(this.fieldConfig, {name:'name'}).tooltip = "Dataset name (read-only)."
     }
-
+    this.entityForm.formGroup.controls['recordsize'].valueChanges.subscribe((res)=>{
+      const res_number = parseInt(this.reverseRecordSizeMap[res],10);
+      if(this.recommended_size){
+        const recommended_size_number = parseInt(this.reverseRecordSizeMap[this.recommended_size],0);
+        if (res_number < recommended_size_number){
+          _.find(this.fieldConfig, {name:'recordsize'}).warnings = `
+          Based on Pool Topology, it's highly recommnded to set Record Size to ${this.recommended_size}. 
```

Comment:
Typo, odd capitalization, and not clear.  Suggested exact replacement wording:
`Recommended record size based on pool topology: ${this.recommended_size}.`